app/components/text-currency/text-currency.tsx

```diff
@@ -30,8 +31,11 @@ export const TextCurrencyForAmount: ComponentType = ({
   iconColor = palette.black,
 }: Props) => {
   if (currency === "USD") {
-    const amountDisplay = Number.isNaN(amount) ? "..." : usdAmountDisplay(amount)
-    return <Text style={style}>{amountDisplay}</Text>
+    return (
+      <Text style={style}>
```

Comment:
Here you are nesting DisplayCurrency in a Text tag, even though it is already a Text element. Would it make sense to have DisplayCurrency accept the text style, or have it not be a component at all but instead be a hook that returns the raw text value?

Reply:
I've removed this and it's now using the function to generate the display currency.